docs: Add adrv9361z7035 doc (#1609)
analogdevicesinc/hdl: docs/projects/adrv9361z7035/index.rst

.. _adrv9361z7035:

ADRV9361Z7035 HDL Project
===============================================================================

Overview
-------------------------------------------------------------------------------

The :adi:`ADRV9361-Z7035` is built on a portfolio of highly integrated

> System-On-Module (SOMs) based on the Xilinx Zynq®-7000 All Programmable (AP)SoC.

Reviewer: Not entirely sure about the sign, if it will work in the PDF doc., so better to remove it.
```suggestion
System-On-Module (SOMs) based on the Xilinx Zynq-7000 All Programmable (AP) SoC.
```

Starting with the :adi:`AD9361`, it is schematically & HDL similar to the
:ref:`fmcomms2`.

> The purpose of the :adi:1ADRV9361-Z70351 RF SOM is to provide an RF platform to

Reviewer: ```suggestion
The purpose of the :adi:`ADRV9361-Z7035` RF SOM is to provide an RF platform to
```

   ccbob_cmos    ADRV9361Z7035-SOM (CMOS Mode) + ADRV1CRR-BOB
   ccbob_lvds    ADRV9361Z7035-SOM (LVDS Mode) + ADRV1CRR-BOB
   ccpackrf_lvds ADRV9361Z7035-SOM (LVDS Mode) + ADRV1CRR-PACKRF
   ccfmc_lvds    ADRV9361Z7035-SOM (LVDS Mode) + ADRV1CRR-FMC
   ccpci_lvds    ADRV9361Z7035-SOM (LVDS Mode) + ADRV1CRR-PCI
   ccusb_lvds    ADRV9361Z7035-SOM (LVDS Mode) + ADRV1CRR-USB
   ============= ===============================================
   
.. collapsible:: Board Design Files (Vivado IPI)

   =========================== =====================================
   Directory/File              Description
   =========================== =====================================
   common/adrv9361z7035_bd.tcl ADRV9361Z7035-SOM board design file.
   common/ccbob_bd.tcl         carrier, break out board design file.
   common/ccpackrf_bd.tcl      carrier, pack rf board design file.
   common/ccfmc_bd.tcl         carrier, fmc board design file.
   common/ccpci_bd.tcl         carrier, pci-e board design file.
   common/ccusb_bd.tcl         carrier, usb board design file.
   =========================== =====================================
   
   .. note::
   

>       FMC & BOB carrier designs includes loopback daughtercards for connectivity

Reviewer: ```suggestion
      FMC & BOB carrier designs include loopback daughtercards for connectivity
```

   | adrv9361z7035_constr_lvds.xdc | constraints file.                    |
   +-------------------------------+--------------------------------------+
   | common/ccbob_constr.xdc       | carrier, break out board constraints |
   |                               | file.                                |
   +-------------------------------+--------------------------------------+
   | common/ccpackrf_constr.xdc    | carrier, packrf board constraints    |
   |                               | file.                                |
   +-------------------------------+--------------------------------------+
   | common/ccfmc_constr.xdc       | carrier, fmc board constraints file. |
   +-------------------------------+--------------------------------------+
   | common/ccpci_constr.xdc       | carrier, pci-e board constraints     |
   |                               | file.                                |
   +-------------------------------+--------------------------------------+
   | common/ccusb_constr.xdc       | carrier, usb board constraints file. |
   +-------------------------------+--------------------------------------+
   
   .. note::
   

>       FMC & BOB carrier designs includes loopback daughtercards for connectivity

Reviewer: ```suggestion
      FMC & BOB carrier designs include loopback daughtercards for connectivity
```

      testing.

A more comprehensive build guide can be found in the :ref:`build_hdl` user guide.

Resources
-------------------------------------------------------------------------------

Systems related
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Here you can find the quick start guides available for these evaluation boards:
> 
> - :dokuwiki:`ADRV9361-Z7035 <resources/eval/user-guides/adrv9361-z7035>`

Reviewer: ```suggestion
- :dokuwiki:`ADRV9361-Z7035 quick start guide<resources/eval/user-guides/adrv9361-z7035>`
```

   :header-rows: 1

   * - IP name
     - Source code link
     - Documentation link
   * - AXI_AD9361
     - :git-hdl:`library/axi_ad9361`
     - :ref:`axi_ad9361`
   * - AXI_CLKGEN
     - :git-hdl:`library/axi_clkgen` *
     - :ref:`axi_clkgen`
   * - AXI_DMAC
     - :git-hdl:`library/axi_dmac`
     - :ref:`axi_dmac`
   * - AXI_GPREG
     - :git-hdl:`library/axi_gpreg` **
     - ---
   * - AXI_HDMI_TX
     - :git-hdl:`library/axi_hdmi_tx` *
     - :ref:`axi_hdmi_tx`
   * - AXI_I2S_ADI
     - :git-hdl:`library/axi_i2s_adi` \***
     - ---
   * - AXI_SPDIF_TX
     - :git-hdl:`library/axi_spdif_tx` *
     - ---
   * - AXI_SYSID
     - :git-hdl:`library/axi_sysid`
     - :ref:`axi_sysid`
   * - SYSID_ROM
     - :git-hdl:`library/sysid_rom`
     - :ref:`axi_sysid`
   * - UTIL_CPACK2
     - :git-hdl:`library/util_pack/util_cpack2`
     - :ref:`util_cpack2`
   * - UTIL_RFIFO
     - :git-hdl:`library/util_rfifo`
     - :ref:`util_rfifo`
   * - UTIL_TDD_SYNC
     - :git-hdl:`library/util_tdd_sync`
     - ---
   * - UTIL_UPACK2
     - :git-hdl:`library/util_pack/util_upack2`
     - :ref:`util_upack2`
   * - UTIL_WFIFO
     - :git-hdl:`library/util_wfifo`
     - :ref:`util_wfifo`

>    * - XILINX_AXI_XCVRLB
>      - :git-hdl:`library/xilinx/axi_xcvrlb` **
>      - ---
>    * - XILINX_UTIL_CLKDIV

Reviewer: ```suggestion
   * - AXI_XCVRLB
     - :git-hdl:`library/xilinx/axi_xcvrlb` **
     - ---
   * - UTIL_CLKDIV
```

     - :git-hdl:`library/xilinx/util_clkdiv`
     - ---

.. admonition:: Legend
   :class: note

   - ``*`` only used for ccfmc_lvds (ADRV1CRR-FMC)  
   - ``**`` only used for ccbob_cmos (ADRV1CRR-BOB), ccbob_lvds (ADRV1CRR-BOB), ccfmc_lvds (ADRV1CRR-FMC)  
   - ``***`` only used for ccfmc_lvds (ADRV1CRR-FMC), ccpackrf_lvds (ADRV1CRR-PACKRF)  

Software related
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

- :git-linux:`ADRV9361-Z7035 ccbob_cmos Linux device tree <arch/arm/boot/dts/xilinx/zynq-adrv9361-z7035-bob-cmos.dts>`
- :git-linux:`ADRV9361-Z7035 ccbob_lvds Linux device tree <arch/arm/boot/dts/xilinx/zynq-adrv9361-z7035-bob.dts>`
- :git-linux:`ADRV9361-Z7035 ccfmc_lvds Linux device tree <arch/arm/boot/dts/xilinx/zynq-adrv9361-z7035-fmc.dts>`
- :git-linux:`ADRV9361-Z7035 ccfmc_lvds Linux device tree <arch/arm/boot/dts/xilinx/zynq-adrv9361-z7035-userspace.dts>`
- :git-linux:`ADRV9361-Z7035 ccfmc_lvds Linux device tree - TDD mode <arch/arm/boot/dts/xilinx/zynq-adrv9361-z7035-fmc-rfcard-tdd.dts>`
- :git-linux:`ADRV9361-Z7035 ccpackrf_lvds Linux device tree <arch/arm/boot/dts/xilinx/zynq-adrv9361-z7035-packrf.dts>`

> - :dokuwiki:`[Wiki]  ADRV9361-Z7035 User Guide - Software <resources/eval/user-guides/adrv936x_rfsom/user-guide/software>`
> - :dokuwiki:`[Wiki]  AD9361 Linux device driver <resources/tools-software/linux-drivers/iio-transceiver/ad9361>`

Reviewer: ```suggestion
- :dokuwiki:`[Wiki] ADRV9361-Z7035 User Guide - Software documentation<resources/eval/user-guides/adrv936x_rfsom/user-guide/software>`
- :dokuwiki:`[Wiki] AD9361 Linux device driver documentation <resources/tools-software/linux-drivers/iio-transceiver/ad9361>`
```
